### PR TITLE
Using gem ransack to search and sort user

### PR DESCRIPTION
**Pull request by haonh**

### CODE REVIEW

File: app/helpers/application_helper.rb [hunk 1]
```diff
@@ -3,4 +3,13 @@ def full_title page_title = ""
...
     page_title.empty? ? base_title : page_title + " | " + base_title
   end
+
+  def link_to_add_fields name, f, type
+    new_object = f.object.send "build_#{type}"
+    id = "new_#{type}"
+    fields = f.send("#{type}_fields", new_object, child_index: id) do |builder|
+      render(type.to_s + "_fields", f: builder)
+    end
+    link_to(name, '#', class: "add_fields", data: {id: id, fields: fields.gsub("\n", "")})
```
Comment (hienbx94) [inline app/helpers/application_helper.rb]: '

Comment (trantube94) [inline app/helpers/application_helper.rb]: ''

Comment (thiennv57) [inline app/helpers/application_helper.rb]: ' ' => " "

File: app/views/admin/users/_search_form.html.erb [hunk 2]
```diff
@@ -0,0 +1,18 @@
+<%= search_form_for [:admin, @search] do |f| %>
+  <%= f.condition_fields do |c| %>
+    <%= render "condition_fields", f: c %>
+  <% end %>
+  <p><%= link_to_add_fields t("admin.user.add_condition"), f, :condition %>
```
Comment (ttuan) [inline app/views/admin/users/_search_form.html.erb]: đóng thẻ </p>

**Pull request merged**